krak-rent-frontend/src/pages/SearchResultsPage.js

```diff
@@ -1,21 +1,85 @@
-import React, { Component } from 'react';
-import Axios from 'axios'
+import React, { Component } from "react";
+import Axios from "axios";
+import GeneralAnnouncementContainer from "../components/GeneralAnnouncementContainer";
+import PaginationBar from "../components/PaginationBar";
+import "../css/search-results.css";
 
 class SearchResultsPage extends Component {
-    componentDidMount() {
-        console.log(this.props.location.state.searchJsonBody)
-        const config = {headers: {'Content-Type': 'application/json'}};
-        Axios.post('announcements/get/search-results', this.props.location.state.searchJsonBody, config)
-            .then(
-                results => console.log(results.data.content)
-            )
-    }
+  constructor(props) {
+    super(props);
+    this.state = {
+      announcements: [],
+      pageSize: 10,
+      totalPages: 0,
+      activePage: 0
+    };
+  }
+
+  componentDidMount() {
+    console.log(this.props.location.state.searchJsonBody);
+    const config = { headers: { "Content-Type": "application/json" } };
+    Axios.post(
```

Comment:
POST request should be extracted to the method - remove duplicated code, please

Reply:
Moved to method. Still less code could be written but in that way it's more reusable.